feed/obs/thread.js

```diff
@@ -19,20 +21,23 @@ exports.create = function (api) {
     if (!ref.isLink(rootId)) throw new Error('an id must be specified')
     var sync = Value(false)
 
+    const blocking = api.contact.obs.blocking(api.keys.sync.id())
+
     var prepend = MutantArray()
     api.sbot.async.get(rootId, (err, value) => {
       sync.set(true)
       if (!err) {
-        prepend.push(
-          Value(unboxIfNeeded({key: rootId, value}))
-        )
+        var msg = unboxIfNeeded({key: rootId, value})
+        if (blocking().includes(msg.value.author)) msg.isBlocked = true
+        prepend.push(Value(msg))
       }
     })
 
     var backlinks = api.backlinks.obs.for(rootId)
     var replies = map(computed(backlinks, (msgs) => {
       return msgs.filter(msg => {
-        return msg.value.content.type !== 'vote' && (
+        return !blocking().includes(msg.value.author) &&
+          msg.value.content.type !== 'vote' && (
           api.message.sync.root(msg) === rootId ||
           matchAny(msg.value.content.branch, rootId)
         )
```

Comment:
[my style] : when logic starts getting long to read, I usually want to prune it down so that I (nor others) won't trip over it.
Something like this is a small tweak that might help make it easier to read

```js
const { content, author } = msg.value     // reduce repetition needed in logic

return content.type !== 'vote'      // put easy to test + fail cases first (js will quit on the first `false`)
  && !blocking.includes(author) 
  && (api.message.sync.root(msg) === rootId || matchAny(content.branch, rootId))

```